### PR TITLE
Add requirement pins

### PR DESCRIPTION
**Pull request by gerrod3**
[noissue]
@bmbouter dependabot will update the requirements appropriately or do I need to change something?

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-pulpcore>=3.18
-PyYAML
-semantic_version
+aiofiles>=0.8.0,<0.9
+async_lru>=1.0,<1.1
+galaxy_importer>=0.4.5,<0.5
+GitPython>=3.1.24,<3.2
+jsonschema>=4.4,<4.7
+packaging>=21.3,<22
+pulpcore>=3.20
```
Comment (mdellweg) [inline requirements.txt:7]: I think, with the new deprecation policy, we should specify `pulpcore>=3.20,<3.25` here. The reason we kept this without the upper bound was that we had to change it with every pulpcore release. And that very reason is addressed by the new policy.

Comment (bmbouter) [reply requirements.txt:7]: I agree. @mdellweg how can we identify this more broadly for all plugin writers?

**Comment by bmbouter** [comment]
I reran the CI job twice and it failed on that 1 test, 1 job both times. I don't know why.

File: requirements.txt (hunk 1 above)
Comment (mdellweg) [reply requirements.txt]: We could move the upper bounds check from the release pipeline to the main ci.

**Review by mdellweg — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (bmbouter) [reply requirements.txt]: We should do this @dkliban @fao89 FYI. Basically we want to address release-time issues ahead of release time FTW.

Comment (fao89) [reply requirements.txt]: it is a matter of calling https://github.com/pulp/pulp_ansible/blob/main/.ci/scripts/upper_bound.py at lint time

Comment (bmbouter) [reply requirements.txt]: @fao89 how would you feel about making that PR?

Comment (fao89) [reply requirements.txt]: I can do it, but people need to be aware as it would break all the CIs

Comment (fao89) [reply requirements.txt]: https://github.com/pulp/plugin_template/pull/666

**Comment by fao89** [comment]
is anything blocking the merge?

**Pull request merged**